runtime-modules/storage/src/data_directory.rs

```diff
@@ -279,13 +295,32 @@ decl_module! {
             <KnownContentIds<T>>::put(upd_content_ids);
         }
 
-        /// Sets the content id from the list of known content ids. Requires root privileges.
-        fn set_known_content_id(origin, content_ids: Vec<T::ContentId>) {
+        /// Injects a set of data objects and their corresponding content id into the directory.
+        /// The operation is "silent" - no events will be emitted as objects are added.
+        /// The number of objects that can be added per call is limited to prevent the dispatch
+        /// from causing the block production to fail if it takes too much time to process.
+        /// Existing data objects will be overwritten.
+        pub(crate) fn inject_data_objects(origin, objects: DataObjectsMap<T>) {
             ensure_root(origin)?;
 
-            // == MUTATION SAFE ==
+            // limit size - do some benchmarking to test how much we can allow per call.
+            const MAX_OBJECTS: usize = 20;
```

Comment:
This value looks like part of the interface. Probably we should make it configurable and export it using Substrate.

Reply:
I was really trying to limit the change to the runtime api as much as possible to avoid adding an technical debt, since this method is a one time helper method, but I see your point. I'll export it as a constant.

Reply:
fixed in 8d88c3823285ff4889109385424e36156c059812